File: Part 2 - Regression/Section 4 - Simple Linear Regression/Simple_Linear_Regression/simple_linear_reg.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Fri Nov 24 06:35:30 2017

@author: manoj2prabhakar
"""

# Importing the library

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

#importing the dataset
dataset = pd.read_csv('Salary_Data.csv')
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 1].values

#Split dataset into training and test set

from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X,y, test_size=1/3, random_state=0)

#Feature scaling for Age and Salary since the Equilidean distance of Salary will always dominate Age

"""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)"""

#Fitting Simple Linear Regression to training set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train, y_train)

#Predict the test set result

y_pred = regressor.predict(X_test)

#Visualising the test and training set

plt.scatter(X_train, y_train, color = 'red')
plt.plot(X_train, regressor.predict(X_train), color = 'blue')
plt.title('Salary vs Experience of training set')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()

plt.scatter(X_test, y_test, color = 'red')
plt.plot(X_train, regressor.predict(X_train), color = 'blue')
plt.title('Salary vs Experience of test set')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()

File: Part 2 - Regression/Section 4 - Simple Linear Regression/Simple_Linear_Regression/simple_linear_reg.R
# Data Preprocessing

#importing datasheet
dataset = read.csv('Salary_Data.csv')
# dataset = dataset[, 2:3]

#splitting the data into training and test set
install.packages('caTools')

set.seed(123)
split = sample.split(dataset$Salary, SplitRatio = 2/3)
training_set = subset(dataset, split ==TRUE)
test_set = subset(dataset, split ==FALSE)

# training_set[, 2:3] = scale(training_set[, 2:3])
# test_set[, 2:3] = scale(test_set[, 2:3])

#Fitting Simple linear Regression to training set
regressor = lm(formula = Salary ~ YearsExperience,
               data = training_set)

#predict the test set
y_pred = predict(regressor, newdata = test_set)
# install.packages('ggplot2')

ggplot() + 
  geom_point(aes(x = training_set$YearsExperience , y = training_set$Salary), 
             colour = 'red') + 
  geom_line(aes(x = training_set$YearsExperience , y = predict(regressor, newdata = training_set)),
            colour = 'blue') + 
  ggtitle('Salary vs Experience( Training Set )') +
  xlab(' Years of Experience ') +
  ylab(' Salary ')
  

ggplot() + 
  geom_point(aes(x = test_set$YearsExperience , y = test_set$Salary), 
             colour = 'red') + 
  geom_line(aes(x = training_set$YearsExperience , y = predict(regressor, newdata = training_set)),
            colour = 'blue') + 
  ggtitle('Salary vs Experience( Training Set )') +
  xlab(' Years of Experience ') +
  ylab(' Salary ')


